src/cooccur.c

```diff
@@ -301,16 +308,43 @@ int merge_files(int num) {
         remove(filename);
     }
     fprintf(stderr,"\n");
+    free_fid(fid, num);
+    free(pq);
     return 0;
 }
 
+void free_table(HASHREC **ht) {
+    int i;
+    HASHREC* current;
+    HASHREC* tmp;
+    for (i = 0; i < TSIZE; i++) {
+        current = ht[i];
+        while (current != NULL) {
+            tmp = current;
+            current = current->next;
+            free(tmp->word);
+            free(tmp);
+        }
+    }
+    free(ht);
+}
```

Comment:
This looks super familiar.  Can it be refactored with the other change?  As I said, this is my first exposure to this code, so I have no idea how modular it is.

Reply:
My main motivation was solving memory leaks, I didn't want to interfere to much in current implementation, it is a little bit risky in current state. 
Currently there are 4 files and some part of code overlaps in vocab_count.c  and cooccur.c. (I agree that it could be moved to common.c / .h file).

Reply:
Alright, I don't feel strongly about this.  Can worry about reusing code later

Reply:
I can create common.h / common.c file and move there repeating code. Maybe it should be done in separate PR...